src/net/pwendland/javacard/pki/isoapplet/IsoApplet.java

```diff
@@ -184,7 +184,7 @@ protected IsoApplet() {
         rsaPkcs1Cipher = Cipher.getInstance(Cipher.ALG_RSA_PKCS1, false);
 
         try {
-            ecdsaSignature = Signature.getInstance(Signature.ALG_ECDSA_SHA, false);
+            ecdsaSignature = Signature.getInstance((byte)0x91, false);
```

Comment:
```suggestion
            ecdsaSignature = Signature.getInstance(MessageDigest.ALG_NULL, Signature.SIG_CIPHER_ECDSA, Cipher.PAD_NULL, false);
```
This should make it more readable.
We have to update the targetsdk in `build.xml` to `jc304_kit` and import `javacard.security.MessageDigest` in this file.